ipam: azure: Nil pointer dereference in IPAMSuite.TestIpamManyNodes

https://travis-ci.com/github/cilium/cilium/jobs/341572460
```
=== RUN   Test
level=info msg="Synchronized Azure IPAM information" numInstances=2 numSubnets=2 numVirtualNetworks=2 subsys=azure
level=info msg="Synchronized Azure IPAM information" numInstances=2 numSubnets=3 numVirtualNetworks=2 subsys=azure
level=info msg="Synchronized Azure IPAM information" numInstances=100 numSubnets=3 numVirtualNetworks=1 subsys=azure
level=info msg="Discovered new CiliumNode custom resource" name=node0 subsys=ipam
level=info msg="Discovered new CiliumNode custom resource" name=node1 subsys=ipam
level=info msg="Resolving IP deficit of node" available=0 availableForAllocation=256 availableInterfaces=1 instanceID=vm0 maxIPsToAllocate=10 name=node0 neededIPs=10 remainingInterfaces=1 selectedInterface=/subscriptions/xxx/resourceGroups/g1/providers/Microsoft.Compute/virtualMachineScaleSets/vmss11/virtualMachines/vm0/networkInterfaces/vmss11 selectedPoolID=s-1 subsys=ipam used=0
level=info msg="Synchronized Azure IPAM information" numInstances=100 numSubnets=3 numVirtualNetworks=1 subsys=azure
level=info msg="Resolving IP deficit of node" available=0 availableForAllocation=256 availableInterfaces=1 instanceID=vm1 maxIPsToAllocate=10 name=node1 neededIPs=10 remainingInterfaces=1 selectedInterface=/subscriptions/xxx/resourceGroups/g1/providers/Microsoft.Compute/virtualMachineScaleSets/vmss11/virtualMachines/vm1/networkInterfaces/vmss11 selectedPoolID=s-1 subsys=ipam used=0
----------------------------------------------------------------------
PANIC: ipam_test.go:307: IPAMSuite.TestIpamManyNodes
... Panic: runtime error: invalid memory address or nil pointer dereference (PC=0x436355)
/home/travis/.gimme/versions/go1.14.3.linux.amd64/src/runtime/panic.go:969
  in gopanic
/home/travis/.gimme/versions/go1.14.3.linux.amd64/src/runtime/panic.go:212
  in panicmem
/home/travis/.gimme/versions/go1.14.3.linux.amd64/src/runtime/signal_unix.go:695
  in sigpanic
/home/travis/gopath/src/github.com/cilium/cilium/vendor/github.com/sirupsen/logrus/text_formatter.go:264
  in TextFormatter.needsQuoting
/home/travis/gopath/src/github.com/cilium/cilium/vendor/github.com/sirupsen/logrus/text_formatter.go:290
  in TextFormatter.appendValue
/home/travis/gopath/src/github.com/cilium/cilium/vendor/github.com/sirupsen/logrus/text_formatter.go:281
  in TextFormatter.appendKeyValue
/home/travis/gopath/src/github.com/cilium/cilium/vendor/github.com/sirupsen/logrus/text_formatter.go:198
  in TextFormatter.Format
/home/travis/gopath/src/github.com/cilium/cilium/vendor/github.com/sirupsen/logrus/entry.go:255
  in Entry.write
/home/travis/gopath/src/github.com/cilium/cilium/vendor/github.com/sirupsen/logrus/entry.go:231
  in Entry.log
/home/travis/gopath/src/github.com/cilium/cilium/vendor/github.com/sirupsen/logrus/entry.go:268
  in Entry.Log
/home/travis/gopath/src/github.com/cilium/cilium/vendor/github.com/sirupsen/logrus/entry.go:289
  in Entry.Warn
/home/travis/gopath/src/github.com/cilium/cilium/vendor/github.com/sirupsen/logrus/entry.go:293
  in Entry.Warning
node.go:171
  in Node.ResyncInterfacesAndIPs.func1
/home/travis/gopath/src/github.com/cilium/cilium/pkg/azure/types/types.go:178
  in AzureInterface.ForeachAddress
/home/travis/gopath/src/github.com/cilium/cilium/pkg/ipam/types/types.go:323
  in foreachAddress
/home/travis/gopath/src/github.com/cilium/cilium/pkg/ipam/types/types.go:345
  in InstanceMap.ForeachAddress
node.go:158
  in Node.ResyncInterfacesAndIPs
/home/travis/gopath/src/github.com/cilium/cilium/pkg/ipam/node.go:357
  in Node.recalculate
/home/travis/gopath/src/github.com/cilium/cilium/pkg/ipam/node.go:333
  in Node.UpdatedResource
/home/travis/gopath/src/github.com/cilium/cilium/pkg/ipam/node_manager.go:315
  in NodeManager.Update
ipam_test.go:345
  in IPAMSuite.TestIpamManyNodes
/home/travis/.gimme/versions/go1.14.3.linux.amd64/src/reflect/value.go:321
  in Value.Call
/home/travis/.gimme/versions/go1.14.3.linux.amd64/src/runtime/asm_amd64.s:1373
  in goexit
level=info msg="Synchronized Azure IPAM information" numInstances=1 numSubnets=1 numVirtualNetworks=1 subsys=azure
level=info msg="Discovered new CiliumNode custom resource" name=node1 subsys=ipam
level=info msg="Resolving IP deficit of node" available=1 availableForAllocation=255 availableInterfaces=1 instanceID=vm1 maxIPsToAllocate=9 name=node1 neededIPs=9 remainingInterfaces=1 selectedInterface=/subscriptions/xxx/resourceGroups/g1/providers/Microsoft.Compute/virtualMachineScaleSets/vmss11/virtualMachines/vm1/networkInterfaces/vmss11 selectedPoolID=subnet-1 subsys=ipam used=0
level=info msg="Synchronized Azure IPAM information" numInstances=1 numSubnets=1 numVirtualNetworks=1 subsys=azure
level=info msg="Synchronized Azure IPAM information" numInstances=100 numSubnets=3 numVirtualNetworks=1 subsys=azure
level=info msg="Resolving IP deficit of node" available=10 availableForAllocation=246 availableInterfaces=1 instanceID=vm1 maxIPsToAllocate=5 name=node1 neededIPs=5 remainingInterfaces=1 selectedInterface=/subscriptions/xxx/resourceGroups/g1/providers/Microsoft.Compute/virtualMachineScaleSets/vmss11/virtualMachines/vm1/networkInterfaces/vmss11 selectedPoolID=subnet-1 subsys=ipam used=7
level=info msg="Synchronized Azure IPAM information" numInstances=1 numSubnets=1 numVirtualNetworks=1 subsys=azure
level=info msg="Synchronized Azure IPAM information" numInstances=1 numSubnets=1 numVirtualNetworks=1 subsys=azure
https://travis-ci.com/github/cilium/cilium/jobs/341572460
level=info msg="Discovered new CiliumNode custom resource" name=node1 subsys=ipam
level=info msg="Resolving IP deficit of node" available=1 availableForAllocation=255 availableInterfaces=1 instanceID=vm1 maxIPsToAllocate=7 name=node1 neededIPs=7 remainingInterfaces=1 selectedInterface=/subscriptions/xxx/resourceGroups/g1/providers/Microsoft.Compute/virtualMachineScaleSets/vmss11/virtualMachines/vm1/networkInterfaces/vmss11 selectedPoolID=subnet-1 subsys=ipam used=0
level=info msg="Synchronized Azure IPAM information" numInstances=1 numSubnets=1 numVirtualNetworks=1 subsys=azure
level=info msg="Resolving IP deficit of node" available=8 availableForAllocation=248 availableInterfaces=1 instanceID=vm1 maxIPsToAllocate=7 name=node1 neededIPs=7 remainingInterfaces=1 selectedInterface=/subscriptions/xxx/resourceGroups/g1/providers/Microsoft.Compute/virtualMachineScaleSets/vmss11/virtualMachines/vm1/networkInterfaces/vmss11 selectedPoolID=subnet-1 subsys=ipam used=7
level=info msg="Synchronized Azure IPAM information" numInstances=1 numSubnets=1 numVirtualNetworks=1 subsys=azure
OOPS: 4 passed, 1 PANICKED
--- FAIL: Test (0.05s)
```

--- a/pkg/azure/ipam/node.go
+++ b/pkg/azure/ipam/node.go
@@ -158,14 +158,14 @@ func (n *Node) ResyncInterfacesAndIPs(ctx context.Context, scopedLog *logrus.Ent
 	n.manager.instances.ForeachAddress(n.node.InstanceID(), func(instanceID, interfaceID, ip, poolID string, addressObj ipamTypes.Address) error {
 		address, ok := addressObj.(types.AzureAddress)
 		if !ok {
-			log.WithField("ip", ip).Warning("Not an Azure address object, ignoring IP")
+			scopedLog.WithField("ip", ip).Warning("Not an Azure address object, ignoring IP")
 			return nil
 		}
 
 		if address.State == types.StateSucceeded {
 			available[address.IP] = ipamTypes.AllocationIP{Resource: interfaceID}
 		} else {
-			log.WithFields(logrus.Fields{
+			scopedLog.WithFields(logrus.Fields{
 				"ip":    address.IP,
 				"state": address.State,
 			}).Warning("Ignoring potentially available IP due to non-successful state")

